api/v1/existential/getsvg.ts

```diff
@@ -0,0 +1,78 @@
+import { VercelResponse } from "@vercel/node";
+import { ValidationError } from "yup";
+
+import { promiseWithTimeout } from "../../utils/ENSUtils";
+import { ExistentialNFTRequest } from "../../utils/RequestUtils";
+import { fetchTokenData, getMonthlyEtherValue } from "../../utils/TokenUtils";
+import { ExistentialNFTRequestQuerySchema } from "../../utils/ValidationUtils";
+import {
+  getDefaultExistentialNFTSvg,
+  getExistentialNFTSvg,
+} from "../../assets/ExistentialNFTSvg";
+import { getImageBase64Data } from "../../utils/ImageUtils";
+import UAParser from "ua-parser-js";
+
+const TIMEOUT = 9000;
+
+export const handler = async (
+  request: ExistentialNFTRequest,
+  response: VercelResponse
+) => {
+  try {
+    await ExistentialNFTRequestQuerySchema.validate(request.query);
+
+    const {
+      name,
+      chain,
+      symbol: NFTSymbol,
+      token,
+      flowrate,
+      ipfs,
+    } = request.query;
+
+    const productName = name.replace(/\+/g, " ");
+    const tokenAddr = token as string;
+    const monthlyFlowRate = Number(getMonthlyEtherValue(flowrate)).toFixed(2);
+
+    const { symbol: tokenSymbol } = await promiseWithTimeout(
+      fetchTokenData(tokenAddr.toLowerCase(), chain),
+      TIMEOUT
+    );
+
+    const parser = new UAParser(request.headers["user-agent"]);
+    const engine = parser.getEngine();
```

Comment:
What is this used for?

Reply:
to detect the engine the user is rendering the nft with, because it has dynamic elements, and positioning them works different in e.g chrome, and firefox (WebKit, Gecko)